public/js/index.js

```diff
@@ -131,20 +136,23 @@ const createWidget = function (objWidget, objSocket) {
     },
     stop: function (event, ui) {
       const widgetId = event.target.id
-      const widgetPosition = jQuery('#' + widgetId).data('widget').position
+      const widgetData = jQuery('#' + widgetId).getWidgetData()
+      if (!widgetData) return // make sure widget still exists before continuing
```

Comment:
This is mostly a stylistic difference but, in general I don't like comments on the same line for two reasons:
1. tomorrow, when we want to add more logic, we must remember to add a block to the "if" statement
2. it's fairly obvious the "if" statement is testing whether widgetData exists. more informative would be an explanation of why widgetData might not exist and what it means or what to do about it.
I think the following is more clear and less fragile in the future:
```
if (!widgetData) {
  /*
   * The widgetData no longer exists, likely because the widget was deleted from the UI. That condition
   * is acceptable here.
   */
  return
}
```

Reply:
Sounds good, that makes sense